### PR TITLE
Fix formatting none dates

### PR DESCRIPTION
**Pull request by grebessi**
# Description of change
Fix `due_on` being formatted none and formatted as a date.
This bug was found by [this issue](https://github.com/airbytehq/airbyte/issues/3150) on airbyte.

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_formatting_dates.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import unittest
+from unittest import mock
+import singer
+import tap_github.__init__ as tap_github
```
Comment (keu) [inline tests/unittests/test_formatting_dates.py]: it is not clear to me why do we need this weird import?

File: tests/unittests/test_formatting_dates.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        self.json_data = resp
+    
+    def json(self):
+        return [(self.json_data)]
+
+def get_response(json):
+    yield Mockresponse(resp=json)
+
+
+@mock.patch("tap_github.__init__.authed_get_all_pages")
```
Comment (keu) [inline tests/unittests/test_formatting_dates.py]: and here

File: tests/unittests/test_formatting_dates.py (hunk 1 above)
Comment (grebessi) [reply tests/unittests/test_formatting_dates.py]: That was not necessary. I updated it to be more readable.

**Review by keu — APPROVED** [review]
lgtm

**Comment by keu** [comment]
@grebessi  @sherifnada I don't have write access to merge this

**Comment by keu** [comment]
actually, @grebessi I have a question regarding your changes, what will happen in case records don't have `due_to` and we use incremental sync? will we miss these records during sync?

File: tap_github/__init__.py [hunk 3]
```diff
@@ -414,7 +414,7 @@ def get_all_issue_milestones(schemas, repo_path, state, mdata):
                 # the GitHub API doesn't currently allow a ?since param for pulls
                 # once we find the first piece of old data we can return, thanks to
                 # the sorting
-                if bookmark_time and singer.utils.strptime_to_utc(r.get('due_on')) < bookmark_time:
+                if bookmark_time and r.get("due_on") and singer.utils.strptime_to_utc(r.get("due_on")) < bookmark_time:
```
Comment (keu) [inline tap_github/__init__.py]: are we sure that records sorted by `due_on` field?
why can't we use something like `updated`?

**Comment by sherifnada** [comment]
> @grebessi @sherifnada I don't have write access to merge this

@keu you should have admin access now

File: tap_github/__init__.py (hunk 3 above)
Comment (grebessi) [reply tap_github/__init__.py]: I only kept the previous logic. I am also not sure why `due_on` was chosen for this. Should I re-write it with `updated_at` field? There are other fields using `created_at` and `updated_at` which seems more consistent.

Comment (sherifnada) [reply tap_github/__init__.py]: agreed -- created/updated_at seems better here. Could you make this change? We'll need to make sure the cursor field output by the catalog is updated as well.

**Review by sherifnada — APPROVED** [review]
LGTM but could we use updated_at instead?

**Review by keu — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Pull request merged**